### PR TITLE
lnd_test: fixing testGraphTopologyNotifications flake

### PR DESCRIPTION
**Pull request by Crypt-iQ**
This commit fixes a race condition where Alice's gossip filter would be set after the announcement generation between Bob & Carol resulting in Alice receiving no channel updates or node announcements.  Now, Alice and Bob connect before the channel open between Bob and Carol. I also changed the check here:
https://github.com/lightningnetwork/lnd/blob/c4415f04008c70be20b42f85e1ceff3ef99614c2/lnd_test.go#L8609
to `for numChannelUpds < 2 || numNodeAnns < 1` so it actually waits for both channel updates and node announcements. I think it should wait for an additional node announcement from Carol, but that requires a fix of #3058 

This should also be changed in the same manner but would require a fix of #3058 as well:
https://github.com/lightningnetwork/lnd/blob/c4415f04008c70be20b42f85e1ceff3ef99614c2/lnd_test.go#L8460

This flake is identical to #3010

### CODE REVIEW

**Comment by wpaulino** [comment]
Needs a rebase since the tests have moved over to `lntest/itest`.

**Comment by Crypt-iQ** [comment]
rebased @wpaulino

**Comment by Crypt-iQ** [comment]
I think it would be best if this wasn't merged in until #3219 was.  Then I can update the code to properly check for node announcements.

**Comment by Crypt-iQ** [comment]
Rebased since #3219 was merged in

**Comment by halseth** [comment]
@Crypt-iQ Can you update the PR description to explain why this is still needed now that #3219 is merged?

**Comment by Crypt-iQ** [comment]
@halseth #3219 makes sure each node gets the node announcement. There are two problems present in the current `testGraphTopologyNotifications` itest. 

1. It only waits for 2 updates OR 2 node announcements before breaking out of the for loops described in the description. It needs an `||` instead of `&&` to properly wait for both types.
2. There is also a race condition with Alice's gossip filter being set after announcement generation b/t Bob & Carol.

The reason I waited until #3219 was merged in was so that the for loops in case 1 actually receive 2 node announcements.  This way the test actually represents what it says it does.  Should I update the PR description w/ this?

**Comment by wpaulino** [comment]
> There is also a race condition with Alice's gossip filter being set after announcement generation b/t Bob & Carol which causes the flake.

@Crypt-iQ this should actually be fine after all since Alice and Bob are already connected before each test case starts.

**Comment by Crypt-iQ** [comment]
@wpaulino needed because Alice and Bob disconnect with each other here
https://github.com/lightningnetwork/lnd/blob/2711b7c7f622d879c5991968c4be207c55c5e13a/lntest/itest/lnd_test.go#L9021-L9023

File: lntest/itest/lnd_test.go [hunk 1]
```diff
@@ -9031,30 +9031,41 @@ out:
...
-	)
 
 	// Reconnect Alice and Bob. This should result in the nodes syncing up
 	// their respective graph state, with the new addition being the
 	// existence of Carol in the graph, and also the channel between Bob
 	// and Carol. Note that we will also receive a node announcement from
 	// Bob, since a node will update its node announcement after a new
-	// channel is opened.
+	// channel is opened. We connect Alice and Bob here to ensure her
+	// gossip filter isn't set after the announcements between Bob and Carol
```
Comment (halseth) [inline lntest/itest/lnd_test.go]: This kinda changes what this test is checking. Previously it checked that a new node's announcement was synced on connection, now it connects first then the announcement is created.

File: lntest/itest/lnd_test.go [hunk 2]
```diff
@@ -9031,30 +9031,41 @@ out:
...
+			Amt: chanAmt,
+		},
+	)
+
 	// We should receive an update advertising the newly connected node,
 	// Bob's new node announcement, and the channel between Bob and Carol.
 	numNodeAnns = 0
 	numChannelUpds = 0
-	for numChannelUpds < 2 && numNodeAnns < 1 {
+	for numChannelUpds < 2 || numNodeAnns < 2 {
```
Comment (halseth) [inline lntest/itest/lnd_test.go]: 👍 

If you do this (and the other loop condition change) without moving the `openChannel` call, are tests still passing?

Comment (Crypt-iQ) [reply lntest/itest/lnd_test.go]: It passes most of the time, but still flakes

**Comment by halseth** [comment]
@Crypt-iQ can be closed?

**Comment by Roasbeef** [comment]
Still relevant?

**Comment by yyforyongyu** [comment]
replaced by #6825

**Pull request closed without merging**